packages/preview-middleware-client/src/adp/quick-actions/fe-v4/lr-enable-semantic-date-range-filter-bar.ts

```diff
@@ -0,0 +1,60 @@
+import FlexCommand from 'sap/ui/rta/command/FlexCommand';
+import FilterBar from 'sap/ui/mdc/FilterBar';
+
+import { QuickActionContext, SimpleQuickActionDefinition } from '../../../cpe/quick-actions/quick-action-definition';
+import { pageHasControlId } from '../../../cpe/quick-actions/utils';
+import { getControlById } from '../../../utils/core';
+import { executeToggleAction } from './utils';
+import { SimpleQuickActionDefinitionBase } from '../simple-quick-action-base';
+
+export const ENABLE_SEMANTIC_DATE_RANGE = 'enable-semantic-date-range';
+const PROPERTY_PATH = 'controlConfiguration/@com.sap.vocabularies.UI.v1.SelectionFields/useSemanticDateRange';
+const CONTROL_TYPE = 'sap.fe.macros.controls.FilterBar';
+const boolMap: { [key: string]: boolean } = {
+    'true': true,
+    'false': false
+};
+/**
+ * Quick Action for toggling the visibility of "Semantic date range" for filter bar fields in LR.
+ */
+export class ToggleSemanticDateRangeFilterBar
+    extends SimpleQuickActionDefinitionBase
+    implements SimpleQuickActionDefinition
+{
+    constructor(context: QuickActionContext) {
+        super(ENABLE_SEMANTIC_DATE_RANGE, [], '', context);
+    }
+    readonly forceRefreshAfterExecution = true;
+    private isUseDateRangeTypeEnabled = false;
+
+    initialize(): void {
+        const controls = this.context.controlIndex[CONTROL_TYPE] ?? [];
+        for (const control of controls) {
+            const isActionApplicable = pageHasControlId(this.context.view, control.controlId);
+            const filterBar = getControlById<FilterBar>(control.controlId);
+            if (isActionApplicable && filterBar) {
+                this.control = filterBar;
+                this.isUseDateRangeTypeEnabled = boolMap[this.control.data('useSemanticDateRange')];
+            }
+        }
+    }
+
+    protected get textKey() {
+        return this.isUseDateRangeTypeEnabled
+            ? 'V4_QUICK_ACTION_LR_DISABLE_SEMANTIC_DATE_RANGE_FILTER_BAR'
+            : 'V4_QUICK_ACTION_LR_ENABLE_SEMANTIC_DATE_RANGE_FILTER_BAR';
+    }
+
+    async execute(): Promise<FlexCommand[]> {
+        const command = await executeToggleAction(
```

Comment:
But here check for this.control is missing. Is it correct?

Reply:
its one here: https://github.com/SAP/open-ux-tools/pull/2510/files#diff-32c7a43eab61ae4ec06ccc82d931b96598f97be2b98ea0f8401fa11b88b6364eR60